src/osg/TextureImage.cpp

```diff
@@ -0,0 +1,23 @@
+#include <osg/TextureImage>
+
+using namespace osg;
+
+int ImageTextureBinding::compare(const osg::StateAttribute &sa)const{
+    COMPARE_StateAttribute_Types(ImageTextureBinding,sa)
+    // Compare each parameter in turn against the rhs.
+    COMPARE_StateAttribute_Parameter(_target)
+    COMPARE_StateAttribute_Parameter(_imageunit)
+    COMPARE_StateAttribute_Parameter(_access)
+    COMPARE_StateAttribute_Parameter(_format)
+    COMPARE_StateAttribute_Parameter(_layered)
+    COMPARE_StateAttribute_Parameter(_level)
+    return 0;
+}
+
+void ImageTextureBinding::apply(osg::State&state)const{
+    if(_target.valid()){
+      osg::Texture::TextureObject *to= _target->getTextureObject( state.getContextID() );
```

Comment:
Standard StateAttribute in a StateSet will be applied by the OSG before a StateSet's Texture atttributes so there will be an order of creation/apply issue with using the TextureObject of a Texture. as this could end up being required before the actual texture is applied. 

I have no answer for this issue as I have only do a quick code review trying to understand the feature, but this one stands out as a red flag that would need resolving somehow.

Reply:
I understand the issue
I originally had a target.apply before getTextureObject but removing it still give good result with my complex test case (Sparse Voxel Octree Voxelization) ...:/
will investigate if targets are applied in some way....